packages/jetpack-ai-calypso/src/logo-generator/components/generator-modal.tsx

```diff
@@ -29,32 +31,69 @@ export const GeneratorModal: React.FC< GeneratorModalProps > = ( {
 	onClose,
 	siteDetails,
 } ) => {
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	const { setSiteDetails, fetchAiAssistantFeature } = useDispatch( STORE_NAME );
-	const [ isLoading, setIsLoading ] = useState( true );
-	const { selectedLogo } = useLogoGenerator();
+	const [ loadingState, setLoadingState ] = useState<
+		'loadingFeature' | 'analyzing' | 'generating' | null
+	>( null );
+	const [ initialPrompt, setInitialPrompt ] = useState< string | undefined >();
+	const [ isFirstCallOnOpen, setIsFirstCallOnOpen ] = useState( true );
```

Comment:
I'd change this to only depend on an empty history. As this is, refreshing (hard) the page and reopening the modal would request a new logo (I think)